python_modules/libraries/dagster-airbyte/dagster_airbyte/asset_decorator.py

```diff
@@ -0,0 +1,312 @@
+from functools import partial
+from itertools import chain
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    Iterable,
+    List,
+    Mapping,
+    Optional,
+    Sequence,
+    Set,
+    Union,
+    cast,
+)
+
+from dagster import (
+    AssetDep,
+    AssetKey,
+    AssetOut,
+    AssetsDefinition,
+    AutoMaterializePolicy,
+    DagsterInvalidDefinitionError,
+    FreshnessPolicy,
+    Nothing,
+    SourceAsset,
+    TableSchema,
+    _check as check,
+    multi_asset,
+)
+from dagster._core.definitions.events import CoercibleToAssetKey
+from dagster._core.definitions.metadata import MetadataValue, TableSchemaMetadataValue
+from dagster._utils.warnings import suppress_dagster_warnings
+
+from dagster_airbyte.airbyte_project import AirbyteProject
+from dagster_airbyte.dagster_airbyte_translator import DagsterAirbyteTranslator
+from dagster_airbyte.utils import get_schema_by_table_name, table_to_output_name_fn
+
+
+# TODO: pass all connections, then add connection filter or selection
+# TODO: add bool to materialize normalization table
+# TODO: add DagsterAirbyteTranslator
+@suppress_dagster_warnings
+def airbyte_single_connection_asset(
+    *,
+    connection_id: str,
+    destination_tables: Sequence[str],
+    asset_key_prefix: Optional[Sequence[str]] = None,
+    group_name: Optional[str] = None,
+    io_manager_key: Optional[str] = None,
+    normalization_tables: Optional[Mapping[str, Set[str]]] = None,
+    deps: Optional[Iterable[Union[CoercibleToAssetKey, AssetsDefinition, SourceAsset]]] = None,
+    upstream_assets: Optional[Set[AssetKey]] = None,
+    schema_by_table_name: Optional[Mapping[str, TableSchema]] = None,
+    freshness_policy: Optional[FreshnessPolicy] = None,
+    stream_to_asset_map: Optional[Mapping[str, str]] = None,
+    auto_materialize_policy: Optional[AutoMaterializePolicy] = None,
+    dagster_airbyte_translator: Optional[DagsterAirbyteTranslator] = None,
```

Comment:
would prefer to use a single `AssetSpec` here rather than exploding out all the parameters

Reply:
I agree completely - the `airbyte_single_connection_asset` asset decorator was the initial test, derived from the current `build_airbyte_assets`. 

Most likely, this approach will not be adopted, in favour of the `airbyte_assets` asset decorator, derived from `load_assets_from_airbyte_resource`. This asset will represent all connections in Airbyte and allow users to select which connections to materialize, similar to what we do with `dbt_assets`. The plan is to use `AssetSpec` for `airbyte_assets`.